### ISSUE TITLE
Barrels now display multiple items on top of the barrel

### ISSUE THREAD

**Issue by pmyers123** [John000708/Barrels#30]
After placing a barrel of any size down, the barrel starts to add multiple items of either what is in the barrel on top of it, or the barrel adds the barrier sign over and over if no items are in the barrel.

Here's the screenshot

![image](https://user-images.githubusercontent.com/40214829/88209224-ff6eff80-cc17-11ea-9218-9e388cfb0b7c.png)

This should not happen. All we should see on top is one single item with the counter

Here's the slimefun version info

![image](https://user-images.githubusercontent.com/40214829/88209277-17df1a00-cc18-11ea-83f2-61a3eaa404f4.png)

**Comment by mrcoffee1026** [comment]
I usually only saw this when breaking a barrel but this happens.  Eventually there are so many of the floating, titled, undestroyable item that the server simply lags to death.  You've got to be quick with the kill command in this location to get rid of them before the server crashes... and of course you have to actually use your debug fish to eliminate the sf data from the blocks that were in use by the barrels in order for it to stop recurring.  Of all issues with barrel, this is the one making it questionable to continue use.  I already have to advise players to not break them, but if it can happen at any time on unbroken barrels then there's just no stopping it from becoming a problem at some point or another.

**Comment by pmyers123** [comment]
@mrcoffee1026 have you found a solution to stop the problem but allow the barrel to be used, since shift left click with the debug fish turns it back into an acacia log for all practical purposes? 

This is a relatively new bug from what I have seen. Dev's, if there's any additional info i can provide to help resolve. I will attach the server error logs for when this happened.

[D.zip](https://github.com/John000708/Barrels/files/4969103/D.zip)

**Comment by mrcoffee1026** [comment]
No not a solution to stop it.  Just a way to get it to stop after the blocks are broken.  This is not a new bug in my experience, this has been going on for ages, although usually only after the barrel has already been broken, if it's happening before the barrels break now, then this addon is truly doomed.  You still have to break the barrel.  Use Creative mode if you have to.  Once the block is broken you can - in creative mode -  shift right click with the debug fish to place a placeholder block where the barrel was and the item marker was above it... then shift left click with the fish on both of these blocks to erase the slimefun data in those locations to have the plugin cease to generate the never ending display items at that location, then punch each of the placeholder blocks into oblivion, switch back to survival and place down new barrels.

**Comment by mrcoffee1026** [comment]
yeah does display multiple items (i count 4 on everything even though when "killing" the remaining item, those 4 only counts as a single entity.  seems to be a direct result of the display item now perfectly matching the stored item (it didn't used to since the display item contained invalid color codes to hide the text "barrel".  This costs you an extra stack of items per barrel since the display items can't actually be removed, and they're still left behind as an un pickup-able item when the barrel is broken, although I'm now only seeing my other glitch very sporadically and it seems rare that breaking a barrel causes an infinite pile of named unretrievable items to accumulate in that location.  Of issues, this is the less problematic one, but you still have to use a command to kill the floating item after you break the barrel like /minecraft:kill @e[type=item,distance=..2]

**Comment by pmyers123** [comment]
This was not fixed in version 13 or 14. Any updates on this?

**Comment by mrcoffee1026** [comment]
I no longer see multiple items on top of barrels at any point, even when they're broken later.  The only real problem now is that when a barrel is broken, a barrel is not dropped.  Instead, the equivalent vanilla block is dropped along with the display item which cannot be picked up.  That item will eventually despawn on its own but if it does there is no evidence that they'd ever placed a barrel... and of course they'll want their barrel replaced.

**Comment by pmyers123** [comment]
They are still there, even on the latest version. I've got video proof of it. On version 14 that came out Aug 5, 2020.  Version 13 did not have the fix either.

i found a temporary work around until the code is actually fixed. In the config.yml file, if you turn the display items option to false, you don't see the hundreds of items piling up above the barrels.. However you now don't see what's in there, or any count of the items unless you open the GUI of the barrel. I'm not having any problems picking up the barrels when mined. All of my plugins for slimefun are 100% up to date on the latest versions as of today.

**Comment by mrcoffee1026** [comment]
I'm going to attempt your workaround and see if that also fixes the breaking/replacing them issue we have.  I'm fine with disabling the items anyway since it's just entity lag I don't really want anyway, nevermind how it used to spawn millions of the items from time to time, this could save a lot of trouble, I hadn't realized it was an option.
